libres/lib/res_util/block_fs.cpp

```diff
@@ -130,8 +133,7 @@ struct block_fs_struct {
     int block_size;          /* The size of blocks in bytes. */
     int lock_fd; /* The file descriptor for the lock_file. Set to -1 if we do not have write access. */
 
-    pthread_mutex_t io_lock;  /* Lock held during fread of the data file. */
-    pthread_rwlock_t rw_lock; /* Read-write lock during all access to the fs. */
+    std::mutex mutex;
```

Comment:
Note that a simple mutex is not able to distinguish between write-access and read-access like `pthread_mutex_t` does.

You might want to use a [`shared_mutex`](https://en.cppreference.com/w/cpp/thread/shared_mutex)
and use `lock_shared` for read-access.

OTOH, if you feel this is unnecessary granularity, feel free to argue the fact.

Reply:
I feel this is unnecessary granularity, yes.

`block_fs_has_file` is used only by `ert::block_fs_driver::has_node`, which is only used by `enkf_fs_has_node`, which is used by two tests. But I reckon this is reasonable functionality to keep.

In all other instances it's actually touching the file handle, and you need exclusive access to do this.

Reply:
In other words, when existing code asks for read-only access (`block_fs->io_lock`) it already holds the `rw_lock` and thus has exclusive access. That's fair enough.